dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java

```diff
@@ -1290,8 +1286,58 @@ private void verifyTRSImageConversion(final List<ToolVersion> versions, final St
         assertTrue("Snapshotted version should be in the list", snapshotInList);
     }
 
-    private WorkflowVersion snapshotWorkflowVersion(WorkflowsApi workflowsApi, String workflowPath, String descriptorType, String descriptorPath, String versionName) {
-        Workflow workflow = manualRegisterAndPublish(workflowsApi, workflowPath, "", descriptorType, SourceControl.GITHUB, descriptorPath, true);
+    /**
+     * Test that the image_name is set correctly after TRS image conversion.
+     * This is a separate test from verifyTRSImageConversion because it's difficult to map the snapshot version's images to the
+     * TRS version's images if there's more than 1 Docker reference in the workflow.
+     * This test works with workflows containing 1 Docker reference (may not necessarily have only 1 image because DockerHub can provide
+     * multiple images for a single version, one for each os/architecture).
+     */
+    @Test
+    public void testTRSImageName() {
+        final ApiClient webClient = getWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowsApi = new WorkflowsApi(webClient);
+        final io.dockstore.openapi.client.ApiClient openAPIClient = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        Ga4Ghv20Api ga4Ghv20Api = new Ga4Ghv20Api(openAPIClient);
+        WorkflowVersion snapshotVersion;
+        ToolVersion trsVersion;
+
+        Workflow workflow = manualRegisterAndPublish(workflowsApi, "dockstore-testing/hello-wdl-workflow", "",
+                DescriptorType.WDL.toString(), SourceControl.GITHUB, "/Dockstore.wdl", true);
+
+        // Workflow with Quay image specified using a tag
+        String quayTagVersionName = "1.0";
+        snapshotVersion = snapshotWorkflowVersion(workflowsApi, workflow, quayTagVersionName);
+        assertEquals("Should only be one image in this workflow", 1, snapshotVersion.getImages().size());
+        trsVersion = ga4Ghv20Api.toolsIdVersionsVersionIdGet("#workflow/github.com/dockstore-testing/hello-wdl-workflow", quayTagVersionName);
```

Comment:
Probably not likely, but this checks number of images in snapshotVersion but not the number of images in the trsVersion. If somehow the conversion died, could the test pass with 0 trs images?

Reply:
Yeah, seems like the test would pass even if the conversion died and there's 0 trs images. Will add a check for the number of images in the trsVersion